lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1316,16 +1316,21 @@ namespace {
 
 class AccessScopeChecker {
   const SourceFile *File;
+  bool TreatUsableFromInlineAsPublic;
   TypeChecker::TypeAccessScopeCacheMap &Cache;
 
 protected:
   ASTContext &Context;
   Optional<AccessScope> Scope = AccessScope::getPublic();
 
   AccessScopeChecker(const DeclContext *useDC,
+                     bool treatUsableFromInlineAsPublic,
```

Comment:
You're definitely starting to convince me that UsableFromInline is really a new access level.